### ISSUE TITLE
Scheduler does not place some DaemonSet-created pods

### ISSUE THREAD

**Issue by seh** [kubernetes/kubernetes#71466]
**What happened**:

I added nodes steadily to my Kubernetes cluster—reaching 200 or so worker nodes, in addition to three masters. As the nodes arrived, I was watching to see how long it took each node to appear after the underlying machine had booted, and then how long it took for the node to become "ready." A handful of nodes—four out of 200—would not become ready. The "calico-node" _DaemonSet_'s pods had not started on those four nodes, and were stuck in the "pending" state. Until the "calico-node" pod starts and provides a CNI plugin, the kubelet won't remove its "node.kubernetes.io/not-ready" taints.

There were four of these pods created for the "calico-node" _DaemonSet_ that were "pending." Each had a node affinity stanza locking it into one and only one candidate node. For example, here's one of the pods seeking assignment to node _ip-10-105-51-241.ec2.internal_:
```yaml
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchFields:
          - key: metadata.name
            operator: In
            values:
            - ip-10-105-51-241.ec2.internal
```
However, the scheduler won't place these pods. They sit unassigned with no "spec.nodeName" field populated. I tried killing the three scheduler pods (one on each master) and allowing them to restart, but they come back up and still don't act on these pods, with no log messages indicating why.

I considered whether the tolerations on these pods couldn't tolerate the taints on the nodes. The four unready nodes are tainted as follows:
```yaml
taints:
- effect: NoSchedule
  key: node.kubernetes.io/not-ready
  timeAdded: 2018-11-27T16:05:47Z
- effect: NoExecute
  key: node.kubernetes.io/not-ready
  timeAdded: 2018-11-27T16:05:52Z
```
The "calico-node" pods are correspondingly tolerant:
```yaml
tolerations:
- effect: NoSchedule
  operator: Exists
- key: CriticalAddonsOnly
  operator: Exists
- effect: NoExecute
  operator: Exists
- effect: NoExecute
  key: node.kubernetes.io/not-ready
  operator: Exists
- effect: NoExecute
  key: node.kubernetes.io/unreachable
  operator: Exists
- effect: NoSchedule
  key: node.kubernetes.io/disk-pressure
  operator: Exists
- effect: NoSchedule
  key: node.kubernetes.io/memory-pressure
  operator: Exists
- effect: NoSchedule
  key: node.kubernetes.io/unschedulable
  operator: Exists
- effect: NoSchedule
  key: node.kubernetes.io/network-unavailable
  operator: Exists
```
Note that there's no explicit match for the "NoSchedule" effect with the "node.kubernetes.io/not-ready" taint key, but there is the wildcard for the "NoSchedule" effect with the "Exists" operator at the top.

To ensure that that omission wasn't the problem, I added the following toleration to the "calico-node" pod:
```yaml
- effect: NoSchedule
  key: node.kubernetes.io/not-ready
  operator: Exists
```
Even with that in place, the scheduler still didn't take any action.

**What you expected to happen**:

The scheduler should act on these pending pods and assign them to the node nominated in their affinity constraint.

**How to reproduce it (as minimally and precisely as possible)**:

Well, this is hard, because it doesn't happen deterministically, so far as I can tell.

Start with a cluster that relies on a CNI plugin that arrives via _DaemonSet_, with a small number of nodes (even no workers beyond the masters). Add nodes in groups of, say, 20 at a time, until you notice that some of the nodes haven't become ready after a few minutes.

Look for pending pods that were created by the _DaemonSet_. Confirm that the target node does exist, but is not yet ready.

**Anything else we need to know?**:

Here are a few pertinent manifests extracted from my cluster.

First, the "calico-node" _DaemonSet_:
<details>

```yaml
apiVersion: extensions/v1beta1
kind: DaemonSet
metadata:
  annotations:
  creationTimestamp: 2018-11-17T23:15:05Z
  generation: 1
  labels:
    k8s-app: calico-node
  name: calico-node
  namespace: kube-system
  resourceVersion: "1868690"
  selfLink: /apis/extensions/v1beta1/namespaces/kube-system/daemonsets/calico-node
  uid: 9e89275c-eabe-11e8-9dec-128456ada3d8
spec:
  revisionHistoryLimit: 10
  selector:
    matchLabels:
      k8s-app: calico-node
  template:
    metadata:
      annotations:
        scheduler.alpha.kubernetes.io/critical-pod: ""
      creationTimestamp: null
      labels:
        k8s-app: calico-node
    spec:
      containers:
      - env:
        - name: DATASTORE_TYPE
          value: kubernetes
        - name: FELIX_LOGSEVERITYSCREEN
          value: info
        - name: CLUSTER_TYPE
          value: k8s,bgp
        - name: CALICO_DISABLE_FILE_LOGGING
          value: "true"
        - name: FELIX_DEFAULTENDPOINTTOHOSTACTION
          value: ACCEPT
        - name: FELIX_IPV6SUPPORT
          value: "false"
        - name: FELIX_IPINIPMTU
          value: "1440"
        - name: WAIT_FOR_DATASTORE
          value: "true"
        - name: CALICO_IPV4POOL_CIDR
          value: 172.16.0.0/13
        - name: CALICO_IPV4POOL_IPIP
          value: CrossSubnet
        - name: FELIX_IPINIPENABLED
          value: "true"
        - name: FELIX_TYPHAK8SSERVICENAME
          valueFrom:
            configMapKeyRef:
              key: typha_service_name
              name: calico-config
        - name: NODENAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: spec.nodeName
        - name: IP
          value: autodetect
        - name: FELIX_HEALTHENABLED
          value: "true"
        image: quay.io/calico/node:v3.1.4
        imagePullPolicy: IfNotPresent
        livenessProbe:
          failureThreshold: 6
          httpGet:
            path: /liveness
            port: 9099
            scheme: HTTP
          initialDelaySeconds: 10
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1
        name: calico-node
        readinessProbe:
          failureThreshold: 3
          httpGet:
            path: /readiness
            port: 9099
            scheme: HTTP
          periodSeconds: 10
          successThreshold: 1
          timeoutSeconds: 1
        resources:
          requests:
            cpu: 250m
        securityContext:
          privileged: true
          procMount: Default
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /lib/modules
          name: lib-modules
          readOnly: true
        - mountPath: /var/run/calico
          name: var-run-calico
        - mountPath: /var/lib/calico
          name: var-lib-calico
      - command:
        - /install-cni.sh
        env:
        - name: CNI_CONF_NAME
          value: 10-calico.conflist
        - name: CNI_NETWORK_CONFIG
          valueFrom:
            configMapKeyRef:
              key: cni_network_config
              name: calico-config
        - name: KUBERNETES_NODE_NAME
          valueFrom:
            fieldRef:
              apiVersion: v1
              fieldPath: spec.nodeName
        image: quay.io/calico/cni:v3.1.4
        imagePullPolicy: IfNotPresent
        name: install-cni
        resources: {}
        terminationMessagePath: /dev/termination-log
        terminationMessagePolicy: File
        volumeMounts:
        - mountPath: /host/opt/cni/bin
          name: cni-bin-dir
        - mountPath: /host/etc/cni/net.d
          name: cni-net-dir
      dnsPolicy: ClusterFirst
      hostNetwork: true
      restartPolicy: Always
      schedulerName: default-scheduler
      securityContext: {}
      serviceAccount: calico-node
      serviceAccountName: calico-node
      terminationGracePeriodSeconds: 0
      tolerations:
      - effect: NoSchedule
        operator: Exists
      - key: CriticalAddonsOnly
        operator: Exists
      - effect: NoExecute
        operator: Exists
      volumes:
      - hostPath:
          path: /lib/modules
          type: ""
        name: lib-modules
      - hostPath:
          path: /var/run/calico
          type: ""
        name: var-run-calico
      - hostPath:
          path: /var/lib/calico
          type: ""
        name: var-lib-calico
      - hostPath:
          path: /opt/cni/bin
          type: ""
        name: cni-bin-dir
      - hostPath:
          path: /etc/cni/net.d
          type: ""
        name: cni-net-dir
  templateGeneration: 1
  updateStrategy:
    rollingUpdate:
      maxUnavailable: 1
    type: RollingUpdate
status:
  currentNumberScheduled: 203
  desiredNumberScheduled: 203
  numberAvailable: 200
  numberMisscheduled: 0
  numberReady: 200
  numberUnavailable: 3
  observedGeneration: 1
  updatedNumberScheduled: 203
```

</details>

Next, one of the pending pods from that _DaemonSet_:
<details>

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    kubernetes.io/psp: privileged
    scheduler.alpha.kubernetes.io/critical-pod: ""
  creationTimestamp: 2018-11-27T15:32:40Z
  generateName: calico-node-
  labels:
    controller-revision-hash: cd68549c4
    k8s-app: calico-node
    pod-template-generation: "1"
  name: calico-node-grc48
  namespace: kube-system
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: DaemonSet
    name: calico-node
    uid: 9e89275c-eabe-11e8-9dec-128456ada3d8
  resourceVersion: "1818402"
  selfLink: /api/v1/namespaces/kube-system/pods/calico-node-grc48
  uid: ad10bba9-f259-11e8-b12a-0e2a5cfd76fc
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchFields:
          - key: metadata.name
            operator: In
            values:
            - ip-10-105-51-241.ec2.internal
  containers:
  - env:
    - name: DATASTORE_TYPE
      value: kubernetes
    - name: FELIX_LOGSEVERITYSCREEN
      value: info
    - name: CLUSTER_TYPE
      value: k8s,bgp
    - name: CALICO_DISABLE_FILE_LOGGING
      value: "true"
    - name: FELIX_DEFAULTENDPOINTTOHOSTACTION
      value: ACCEPT
    - name: FELIX_IPV6SUPPORT
      value: "false"
    - name: FELIX_IPINIPMTU
      value: "1440"
    - name: WAIT_FOR_DATASTORE
      value: "true"
    - name: CALICO_IPV4POOL_CIDR
      value: 172.16.0.0/13
    - name: CALICO_IPV4POOL_IPIP
      value: CrossSubnet
    - name: FELIX_IPINIPENABLED
      value: "true"
    - name: FELIX_TYPHAK8SSERVICENAME
      valueFrom:
        configMapKeyRef:
          key: typha_service_name
          name: calico-config
    - name: NODENAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.nodeName
    - name: IP
      value: autodetect
    - name: FELIX_HEALTHENABLED
      value: "true"
    image: quay.io/calico/node:v3.1.4
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 6
      httpGet:
        path: /liveness
        port: 9099
        scheme: HTTP
      initialDelaySeconds: 10
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    name: calico-node
    readinessProbe:
      failureThreshold: 3
      httpGet:
        path: /readiness
        port: 9099
        scheme: HTTP
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    resources:
      requests:
        cpu: 250m
    securityContext:
      privileged: true
      procMount: Default
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /lib/modules
      name: lib-modules
      readOnly: true
    - mountPath: /var/run/calico
      name: var-run-calico
    - mountPath: /var/lib/calico
      name: var-lib-calico
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: calico-node-token-tsdd2
      readOnly: true
  - command:
    - /install-cni.sh
    env:
    - name: CNI_CONF_NAME
      value: 10-calico.conflist
    - name: CNI_NETWORK_CONFIG
      valueFrom:
        configMapKeyRef:
          key: cni_network_config
          name: calico-config
    - name: KUBERNETES_NODE_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.nodeName
    image: quay.io/calico/cni:v3.1.4
    imagePullPolicy: IfNotPresent
    name: install-cni
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /host/opt/cni/bin
      name: cni-bin-dir
    - mountPath: /host/etc/cni/net.d
      name: cni-net-dir
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: calico-node-token-tsdd2
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  hostNetwork: true
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: calico-node
  serviceAccountName: calico-node
  terminationGracePeriodSeconds: 0
  tolerations:
  - effect: NoSchedule
    operator: Exists
  - key: CriticalAddonsOnly
    operator: Exists
  - effect: NoExecute
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/disk-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/unschedulable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/network-unavailable
    operator: Exists
  volumes:
  - hostPath:
      path: /lib/modules
      type: ""
    name: lib-modules
  - hostPath:
      path: /var/run/calico
      type: ""
    name: var-run-calico
  - hostPath:
      path: /var/lib/calico
      type: ""
    name: var-lib-calico
  - hostPath:
      path: /opt/cni/bin
      type: ""
    name: cni-bin-dir
  - hostPath:
      path: /etc/cni/net.d
      type: ""
    name: cni-net-dir
  - name: calico-node-token-tsdd2
    secret:
      defaultMode: 420
      secretName: calico-node-token-tsdd2
status:
  phase: Pending
  qosClass: Burstable
```

</details>

Finally, here's the _Node_ targeted by that pod:
<details>

```yaml
apiVersion: v1
kind: Pod
metadata:
  annotations:
    kubernetes.io/psp: privileged
    scheduler.alpha.kubernetes.io/critical-pod: ""
  creationTimestamp: 2018-11-27T15:32:40Z
  generateName: calico-node-
  labels:
    controller-revision-hash: cd68549c4
    k8s-app: calico-node
    pod-template-generation: "1"
  name: calico-node-grc48
  namespace: kube-system
  ownerReferences:
  - apiVersion: apps/v1
    blockOwnerDeletion: true
    controller: true
    kind: DaemonSet
    name: calico-node
    uid: 9e89275c-eabe-11e8-9dec-128456ada3d8
  resourceVersion: "1818402"
  selfLink: /api/v1/namespaces/kube-system/pods/calico-node-grc48
  uid: ad10bba9-f259-11e8-b12a-0e2a5cfd76fc
spec:
  affinity:
    nodeAffinity:
      requiredDuringSchedulingIgnoredDuringExecution:
        nodeSelectorTerms:
        - matchFields:
          - key: metadata.name
            operator: In
            values:
            - ip-10-105-51-241.ec2.internal
  containers:
  - env:
    - name: DATASTORE_TYPE
      value: kubernetes
    - name: FELIX_LOGSEVERITYSCREEN
      value: info
    - name: CLUSTER_TYPE
      value: k8s,bgp
    - name: CALICO_DISABLE_FILE_LOGGING
      value: "true"
    - name: FELIX_DEFAULTENDPOINTTOHOSTACTION
      value: ACCEPT
    - name: FELIX_IPV6SUPPORT
      value: "false"
    - name: FELIX_IPINIPMTU
      value: "1440"
    - name: WAIT_FOR_DATASTORE
      value: "true"
    - name: CALICO_IPV4POOL_CIDR
      value: 172.16.0.0/13
    - name: CALICO_IPV4POOL_IPIP
      value: CrossSubnet
    - name: FELIX_IPINIPENABLED
      value: "true"
    - name: FELIX_TYPHAK8SSERVICENAME
      valueFrom:
        configMapKeyRef:
          key: typha_service_name
          name: calico-config
    - name: NODENAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.nodeName
    - name: IP
      value: autodetect
    - name: FELIX_HEALTHENABLED
      value: "true"
    image: quay.io/calico/node:v3.1.4
    imagePullPolicy: IfNotPresent
    livenessProbe:
      failureThreshold: 6
      httpGet:
        path: /liveness
        port: 9099
        scheme: HTTP
      initialDelaySeconds: 10
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    name: calico-node
    readinessProbe:
      failureThreshold: 3
      httpGet:
        path: /readiness
        port: 9099
        scheme: HTTP
      periodSeconds: 10
      successThreshold: 1
      timeoutSeconds: 1
    resources:
      requests:
        cpu: 250m
    securityContext:
      privileged: true
      procMount: Default
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /lib/modules
      name: lib-modules
      readOnly: true
    - mountPath: /var/run/calico
      name: var-run-calico
    - mountPath: /var/lib/calico
      name: var-lib-calico
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: calico-node-token-tsdd2
      readOnly: true
  - command:
    - /install-cni.sh
    env:
    - name: CNI_CONF_NAME
      value: 10-calico.conflist
    - name: CNI_NETWORK_CONFIG
      valueFrom:
        configMapKeyRef:
          key: cni_network_config
          name: calico-config
    - name: KUBERNETES_NODE_NAME
      valueFrom:
        fieldRef:
          apiVersion: v1
          fieldPath: spec.nodeName
    image: quay.io/calico/cni:v3.1.4
    imagePullPolicy: IfNotPresent
    name: install-cni
    resources: {}
    terminationMessagePath: /dev/termination-log
    terminationMessagePolicy: File
    volumeMounts:
    - mountPath: /host/opt/cni/bin
      name: cni-bin-dir
    - mountPath: /host/etc/cni/net.d
      name: cni-net-dir
    - mountPath: /var/run/secrets/kubernetes.io/serviceaccount
      name: calico-node-token-tsdd2
      readOnly: true
  dnsPolicy: ClusterFirst
  enableServiceLinks: true
  hostNetwork: true
  priority: 0
  restartPolicy: Always
  schedulerName: default-scheduler
  securityContext: {}
  serviceAccount: calico-node
  serviceAccountName: calico-node
  terminationGracePeriodSeconds: 0
  tolerations:
  - effect: NoSchedule
    operator: Exists
  - key: CriticalAddonsOnly
    operator: Exists
  - effect: NoExecute
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/not-ready
    operator: Exists
  - effect: NoExecute
    key: node.kubernetes.io/unreachable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/disk-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/memory-pressure
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/unschedulable
    operator: Exists
  - effect: NoSchedule
    key: node.kubernetes.io/network-unavailable
    operator: Exists
  volumes:
  - hostPath:
      path: /lib/modules
      type: ""
    name: lib-modules
  - hostPath:
      path: /var/run/calico
      type: ""
    name: var-run-calico
  - hostPath:
      path: /var/lib/calico
      type: ""
    name: var-lib-calico
  - hostPath:
      path: /opt/cni/bin
      type: ""
    name: cni-bin-dir
  - hostPath:
      path: /etc/cni/net.d
      type: ""
    name: cni-net-dir
  - name: calico-node-token-tsdd2
    secret:
      defaultMode: 420
      secretName: calico-node-token-tsdd2
status:
  phase: Pending
  qosClass: Burstable
```

</details>

If I delete one of these pending pods, the _DaemonSet_ creates a fresh one in its place, and the scheduler places it, causing the node to become "ready." This is when things are "calm" in the cluster, when the node count has stabilized.

**Environment**:
- **Kubernetes version** (use `kubectl version`):
```
Client Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.0-beta.2", GitCommit:"a4ff09c41589c48547e04f85391aa5610ebe0e17", GitTreeState:"clean", BuildDate:"2018-11-23T00:55:34Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"13+", GitVersion:"v1.13.0-beta.2", GitCommit:"a4ff09c41589c48547e04f85391aa5610ebe0e17", GitTreeState:"clean", BuildDate:"2018-11-23T00:47:47Z", GoVersion:"go1.11.2", Compiler:"gc", Platform:"linux/amd64"}
```
- **Cloud provider or hardware configuration**:  
  AWS EC23
- **OS** (e.g. from /etc/os-release):  
```
ID=coreos
VERSION=1911.3.0
VERSION_ID=1911.3.0
BUILD_ID=2018-11-05-1815
PRETTY_NAME="Container Linux by CoreOS 1911.3.0 (Rhyolite)"
ANSI_COLOR="38;5;75"
HOME_URL="https://coreos.com/"
BUG_REPORT_URL="https://issues.coreos.com"
COREOS_BOARD="amd64-usr"
```
- **Kernel** (e.g. `uname -a`):  
  `Linux ip-10-105-51-211 4.14.78-coreos #1 SMP Mon Nov 5 17:42:07 UTC 2018 x86_64 Intel(R) Xeon(R) Platinum 8175M CPU @ 2.50GHz GenuineIntel GNU/Linux`
- Install tools:  
  _kubeadm init_, _kubeadm join_

/kind bug

**Comment by seh** [comment]
Note too that if I delete the unready nodes for which these pods were pending, the pending pods stick around, which sounds like #71349.

**Comment by seh** [comment]
Here are further findings from more testing today, this time with only a single master machine.

The controller manager log file is where all the action takes place. I see the daemon controller complain frequently that it wasn't able to update the status for the "calico-node" and "kube-proxy" _DaemonSets_:
```
daemon_controller.go:304] kube-system/calico-node failed with : error storing status for daemon set &v1.DaemonSet{...}: Operation cannot be fulfilled on daemonsets.apps "calico-node": the object has been modified; please apply your changes to the latest version and try again

node_lifecycle_controller.go:895] Missing timestamp for Node ip-10-105-48-31.ec2.internal. Assuming now as a timestamp.

daemon_controller.go:304] kube-system/kube-proxy failed with : error storing status for daemon set &v1.DaemonSet{...}: Operation cannot be fulfilled on daemonsets.apps "kube-proxy": the object has been modified; please apply your changes to the latest version and try again
```
Which other controllers write to _DaemonSets_? Is this controller racing against itself?

The trouble with the scheduler placing the "calico-node" and "kube-proxy" pods onto new nodes matches with messages like these in the log file:
```
range_allocator.go:314] Failed to update node ip-10-105-53-6.ec2.internal PodCIDR to 172.17.101.0/24 after multiple attempts: failed to patch node CIDR: Node "ip-10-105-53-6.ec2.internal" is invalid: [spec.podCIDR: Forbidden: node updates may not change podCIDR except from "" to valid, []: Forbidden: node updates may only change labels, taints, or capacity (or configSource, if the DynamicKubeletConfig feature gate is enabled)]
range_allocator.go:320] CIDR assignment for node ip-10-105-53-6.ec2.internal failed: failed to patch node CIDR: Node "ip-10-105-53-6.ec2.internal" is invalid: [spec.podCIDR: Forbidden: node updates may not change podCIDR except from "" to valid, []: Forbidden: node updates may only change labels, taints, or capacity (or configSource, if the DynamicKubeletConfig feature gate is enabled)]. Releasing allocated CIDR
event.go:221] Event(v1.ObjectReference{Kind:"Node", Namespace:"", Name:"ip-10-105-53-6.ec2.internal", UID:"0f3a189b-f325-11e8-8047-0aa8995ba554", APIVersion:"", ResourceVersion:"", FieldPath:""}): type: 'Normal' reason: 'CIDRAssignmentFailed' Node ip-10-105-53-6.ec2.internal status is now: CIDRAssignmentFailed
range_allocator.go:314] Failed to update node ip-10-105-53-6.ec2.internal PodCIDR to 172.17.101.0/24 after multiple attempts: failed to patch node CIDR: Node "ip-10-105-53-6.ec2.internal" is invalid: [spec.podCIDR: Forbidden: node updates may not change podCIDR except from "" to valid, []: Forbidden: node updates may only change labels, taints, or capacity (or configSource, if the DynamicKubeletConfig feature gate is enabled)]
range_allocator.go:320] CIDR assignment for node ip-10-105-53-6.ec2.internal failed: failed to patch node CIDR: Node "ip-10-105-53-6.ec2.internal" is invalid: [spec.podCIDR: Forbidden: node updates may not change podCIDR except from "" to valid, []: Forbidden: node updates may only change labels, taints, or capacity (or configSource, if the DynamicKubeletConfig feature gate is enabled)]. Releasing allocated CIDR
range_allocator.go:301] Node ip-10-105-53-6.ec2.internal already has a CIDR allocated 172.17.100.0/24. Releasing the new one 172.17.101.0/24.
```
Those failed _Node_ object updates match the newly arrived nodes that sit unready, with a pair of  "calico-node" and "kube-proxy" pods pending but not placed there by the scheduler. I watched them for about five minutes, during which none of this healed on its own.

I then deleted all the pending "calico-node" and "kube-proxy" pods, and shortly afterward the _DaemonSet_ created replacements and the scheduler placed them on the target nodes, after which the nodes become ready.

**Comment by seh** [comment]
Another test today showed these results: Starting with one node and adding 599 more, only the first 80 became ready, with 520 sitting unready. There were 520 "calico-node" pods pending, and 520 "kube-proxy" pods pending.

I increased the log level on my active "kube-scheduler" container, and inspected the logs. It was grinding away trying to place the "kube-proxy" pods, finding they couldn't tolerate the taints on the one node that would accommodate the pod affinity constraint, and moving on to the next.

Here's the crucial point, though: *it never made it to any of the "calico-node" pods*. Those pods have to land first in order for the node to lose the taints that are repelling the "kube-proxy" pods.

After confirming this problem, I deleted 500 of the pending "kube-proxy" pods. Once I did that, the scheduler made it through all the remaining pending pods, placing the "calico-node" pods, which in turn made the nodes eligible to receive the "kube-proxy" pods.

@bsalamat, does the scheduler only work partway through this queue of pending pods before giving up and starting from the beginning again?

**Comment by seh** [comment]
This problem is likely related to or a duplicate of #71486.

To work around the problem, I'm going to try relaxing the toleration on the "kube-proxy" _DaemonSet_ to allow the scheduler to place those pods more easily, unblocking the pending queue.

**Comment by bsalamat** [comment]
@seh I agree with you that what you are seeing is probably the same as #71486. We are trying to address the two scenarios. One is already merged. The other one is being addressed.

**Comment by seh** [comment]
Having set high priorities on pods that lacked them previously, we haven't seen this problem lately—especially with Kubernetes version 1.13.3.

I'll close this now, and come back if we see the problem arise again.
